script.js

```diff
@@ -1,4 +1,80 @@
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var ownchoice = input.toLowerCase();
+  var computerChoice = random_sps();
+
+  var myOutputValue = "";
+  var sps = "";
+
+  if (computerChoice == 1) {
+    sps = "scissors";
+  }
+
+  if (computerChoice == 2) {
+    sps = "paper";
+  }
+
+  if (computerChoice == 3) {
+    sps = "stone";
+  }
+
+  myOutputValue =
+    win_or_lose(ownchoice, sps) +
+    "<br> <br>" +
+    "Now you can type 'scissors', 'paper' or 'stone' to start a new game. ";
+
+  console.log("You picked", ownchoice);
+  console.log("Computer picked", sps);
+
   return myOutputValue;
 };
+
+var random_sps = function () {
+  var randomNumber = Math.random() * 3;
+  var integernumber = Math.floor(randomNumber);
+  var choices = integernumber + 1;
+  return choices;
+};
+
+var win_or_lose = function (ownchoice, sps) {
+  var msg = "Please select scissors, paper or stone.";
+  if (
+    ownchoice == sps ||
+    (ownchoice == "reversed scissors" && sps == "scissors") ||
+    (ownchoice == "reversed paper" && sps == "paper") ||
+    (ownchoice == "reversed stone" && sps == "stone")
+  ) {
+    msg =
+      "You picked " +
+      ownchoice +
+      " <br> COM picked " +
+      sps +
+      "<br> It's a draw! ";
+    return msg;
+  }
+
+  if (
+    (ownchoice == "scissors" && sps == "paper") ||
+    (ownchoice == "paper" && sps == "stone") ||
+    (ownchoice == "stone" && sps == "scissors") ||
+    (ownchoice == "reversed scissors" && sps == "stone") ||
+    (ownchoice == "reversed paper" && sps == "scissors") ||
+    (ownchoice == "reversed stone" && sps == "paper")
+  ) {
```

Comment:
Notice between line 55 to 62, you have conditionals spanning ~8 lines with a mix of `&&` and `||` operators to check for the win condition.

Likewise, between lines 67 to 74, you have conditions spanning the same number of lines but with sub-conditions slightly flipped, for the loss condition.

**My thought process:**
These 2 `if` conditions are dissimilar, yet also repetitive. Is there a way I can write my logic (and `if` conditions) such that I only have to check either for a win or a loss, but not both?

**How I would do it:**
- I noticed that your default `msg` on line 39 doesn't check for a win/loss/draw condition, but instead checks if the input is valid. Quoting line 39:
> `var msg = "Please select scissors, paper or stone.";`
- I will change it such that the default `msg` now is a win or a loss. When I was testing your code, I used the loss condition as the default, thus:
> `var msg = "You picked " + ownchoice + " <br> COM picked " + sps + "<br> You lost!";`
- Now, that means your input validation has to be handled as a condition, since we are not assuming invalid input as the default anymore. Thus, you can perform your input validation in the first `if`-block, writing something like the quote below. Sure, this `if`-block for input validation is still a little verbose, but at least you do not have separate (and almost repeated, with a mix of `||` and `&&`) conditions that check for loss and win now.
```
if (
    ownchoice != "reversed scissors" &&
    ownchoice != "reversed paper" &&
    ownchoice != "reversed stone" &&
    ownchoice != "scissors" &&
    ownchoice != "paper" &&
    ownchoice != "stone"
  ) {
    msg =
      "Please select 'scissors', 'paper', 'stone', 'reversed scissors', 'reversed paper', or 'reversed stone'.";
  }
```
- Now that you have the default case of loss condition when creating your `msg` variable using `var msg="..."`, we do not need the `if`-block for the loss condition anymore. This means your previous `if`-block for checking losses, the entire thing between lines 67 to 78, can now be deleted. We only need to check for wins now, and we assume if it's not a draw, win, or invalid input, by default it's a loss.